File: go/domain/issueRepository.go
package domain

import (
	"errors"

	"github.com/google/uuid"
)

var IssueNotFoundError = errors.New("Issue not found")

type IssueRepository interface {
	Find(uuid.UUID) (Issue, error)
	FindAll() (issues []Issue, err error)
	Update(Issue) error
	Insert(Issue) error
	Delete(uuid.UUID) error
}

File: node-ts/src/domain/issueRepository.ts
import {Issue} from './issue'

export interface IssueRepository {
	findSingle(id: string) : Promise<Issue>;
	findAll(): Promise<Issue[]>;
	create(issue: Issue): Promise<Issue>
	update(issue: Issue): Promise<Issue>
	delete(id: string): Promise<void>
}

export class NotFoundException{

}